### ISSUE TITLE
utf-8 error

### ISSUE THREAD

**Issue by PUGOSCELIS** [rawdigits/wee-slack#112]
16:47:14     | python: unloading script "slack_extension"
16:47:14     | python: loading script "/home/gromozeka/.weechat/python/autoload/wee_slack.py"
16:47:14     | python: registered script "slack_extension", version 0.98.7 (Extends weechat for typing notification/search/etc on slack.com)
16:47:19     | python: stdout/stderr: Traceback (most recent call last):
16:47:19     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 1753, in url_processor_cb
16:47:19     | python: stdout/stderr:     process_message(message)
16:47:19     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 1439, in process_message
16:47:19     | python: stdout/stderr:     channel.buffer_prnt('unknown', message_json['text'])
16:47:19     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 587, in buffer_prnt
16:47:19     | python: stdout/stderr:     message = message.decode('UTF-8', 'replace')
16:47:19     | python: stdout/stderr:   File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode
16:47:19     | python: stdout/stderr:     return codecs.utf_8_decode(input, errors, True)
16:47:19     | python: stdout/stderr: UnicodeEncodeError: 'ascii' codec can't encode characters in position 120-129: ordinal not in range(128)
16:47:19 =!= | python: error in function "url_processor_cb"
16:47:19     | python: stdout/stderr: Traceback (most recent call last):
16:47:19     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 1753, in url_processor_cb
16:47:19     | python: stdout/stderr:     process_message(message)
16:47:19     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 1439, in process_message
16:47:19     | python: stdout/stderr:     channel.buffer_prnt('unknown', message_json['text'])
16:47:19     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 587, in buffer_prnt
16:47:19     | python: stdout/stderr:     message = message.decode('UTF-8', 'replace')
16:47:19     | python: stdout/stderr:   File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode
16:47:19     | python: stdout/stderr:     return codecs.utf_8_decode(input, errors, True)
16:47:19     | python: stdout/stderr: UnicodeEncodeError: 'ascii' codec can't encode character u'\u2013' in position 166: ordinal not in range(128)
16:47:19 =!= | python: error in function "url_processor_cb"
16:47:19     | python: stdout/stderr: Traceback (most recent call last):
16:47:19     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 1753, in url_processor_cb
16:47:19     | python: stdout/stderr:     process_message(message)
16:47:19     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 1439, in process_message
16:47:19     | python: stdout/stderr:     channel.buffer_prnt('unknown', message_json['text'])
16:47:19     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 587, in buffer_prnt
16:47:19     | python: stdout/stderr:     message = message.decode('UTF-8', 'replace')
16:47:19     | python: stdout/stderr:   File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode
16:47:19     | python: stdout/stderr:     return codecs.utf_8_decode(input, errors, True)
16:47:19     | python: stdout/stderr: UnicodeEncodeError: 'ascii' codec can't encode characters in position 112-117: ordinal not in range(128)
16:47:19 =!= | python: error in function "url_processor_cb"
16:47:20     | python: stdout/stderr: Traceback (most recent call last):
16:47:20     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 1753, in url_processor_cb
16:47:20     | python: stdout/stderr:     process_message(message)
16:47:20     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 1439, in process_message
16:47:20     | python: stdout/stderr:     channel.buffer_prnt('unknown', message_json['text'])
16:47:20     | python: stdout/stderr:   File "/home/gromozeka/.weechat/python/autoload/wee_slack.py", line 587, in buffer_prnt
16:47:20     | python: stdout/stderr:     message = message.decode('UTF-8', 'replace')
16:47:20     | python: stdout/stderr:   File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode
16:47:20     | python: stdout/stderr:     return codecs.utf_8_decode(input, errors, True)
16:47:20     | python: stdout/stderr: UnicodeEncodeError: 'ascii' codec can't encode characters in position 54-59: ordinal not in range(128)
16:47:20 =!= | python: error in function "url_processor_cb"

I comment out line 587, and it earned

**Comment by rawdigits** [comment]
thanks for the stack trace. I'm going to figure out the right way to handle this unexpected state and write some better debug info.

**Comment by rawdigits** [comment]
one request:

try exiting weechat, deleting `~/.weechat/slack.cache` and starting weechat. please let me know if the bug goes away when you do this!

**Comment by bsacharski** [comment]
Getting the same error, removing the `slack.cache` didn't help.

**Comment by roobert** [comment]
+1

Installed the script from master today, here's the log:

```
17:03 script: 299 scripts for WeeChat 1.3
17:03 Plugins loaded: alias, charset, exec, fifo, irc, logger, perl, python, relay, ruby, script, trigger, xfer
17:03 python: stdout/stderr: Traceback (most recent call last):
17:03 python: stdout/stderr:   File "/home/rw/.weechat-work/python/autoload/wee_slack.py", line 1752, in url_processor_cb
17:03 python: stdout/stderr:     process_message(message)
17:03 python: stdout/stderr:   File "/home/rw/.weechat-work/python/autoload/wee_slack.py", line 1439, in process_message
17:03 python: stdout/stderr:     channel.buffer_prnt('unknown', message_json['text'])
17:03 python: stdout/stderr:   File "/home/rw/.weechat-work/python/autoload/wee_slack.py", line 587, in buffer_prnt
17:03 python: stdout/stderr:     message = message.decode('UTF-8', 'replace')
17:03 python: stdout/stderr:   File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode
17:03 python: stdout/stderr:     return codecs.utf_8_decode(input, errors, True)
17:03 python: stdout/stderr: UnicodeEncodeError: 'ascii' codec can't encode character u'\u2019' in position 137: ordinal not in range(128)
17:03 !!! python: error in function "url_processor_cb"
```

**Comment by calve** [comment]
I experienced the same error on startup, ended up with https://github.com/rawdigits/wee-slack/pull/121. I still don't fully understand what happened.

**Comment by calve** [comment]
To elaborate : this exception is raised while trying to print the debug trace. Not that helpfull.

Fix it by applying 5ee8926b8bb1dbd04ebc4cf810bafef3d99ccdcb (so that the Exception handling does not raise an exception before debugging). This will lead you to another UnicodeEncodeError raised by [this line IIRC](https://github.com/rawdigits/wee-slack/blob/master/wee_slack.py#L1510) which should be fixed by #121

**Comment by rawdigits** [comment]
Calve's patch fixed this, and we haven't seen these recur in a long time. Closing this bug.
